crates/sui-swarm-config/src/node_config_builder.rs

```diff
@@ -135,6 +136,41 @@ impl ValidatorConfigBuilder {
         }
     }
 
+    fn build_narwhal_config(
+        for_local_deploy: bool,
+        narwhal_metrics_address: Multiaddr,
+    ) -> narwhal_config::Parameters {
+        let localhost = local_ip_utils::localhost_for_testing();
+        let network_admin_server = NetworkAdminServerParameters {
+            primary_network_admin_server_port: local_ip_utils::get_available_port(&localhost),
+            worker_network_admin_server_base_port: local_ip_utils::get_available_port(&localhost),
+        };
+        let prometheus_metrics = PrometheusMetricsParameters {
+            socket_addr: narwhal_metrics_address,
+        };
+        if for_local_deploy {
+            narwhal_config::Parameters {
+                network_admin_server,
+                prometheus_metrics,
+                ..Default::default()
+            }
+        } else {
+            // If not for local deploy, it must be for testing.
+            narwhal_config::Parameters {
+                network_admin_server,
+                prometheus_metrics,
+                // NOTE: the following parameters are important to ensure tests run fast. Using the default
+                // Narwhal parameters may result in tests taking >60 seconds.
+                header_num_of_batches_threshold: 1,
+                max_header_delay: Duration::from_millis(200),
+                min_header_delay: Duration::from_millis(200),
+                batch_size: 1,
+                max_batch_delay: Duration::from_millis(200),
```

Comment:
`default_max_batch_delay()` is 100ms. We can leave the default since `batch_size` has been specified.